main.go

```diff
@@ -288,13 +300,17 @@ func makeHorizontalPodAutoscalerChanges(kubeClient *k8s.Client, hpa *autoscaling
 			minimumReplicasLowerBound = int32(i)
 		}
 
-		minPodCountBasedOnPrometheusQuery, requestRate, err := getMinPodCountBasedOnPrometheusQuery(kubeClient, hpa, initiator, desiredState)
+		minPodCountBasedOnPrometheusQuery, requestRate, err := getMinPodCountBasedOnPrometheusQuery(kubeClient, hpa, desiredState)
 
 		if err != nil {
 			return status, err
 		}
 
-		minPodCountBasedOnCurrentPodCount := getMinPodCountBasedOnCurrentPodCount(kubeClient, hpa, initiator, desiredState)
+		minPodCountBasedOnCurrentPodCount := minPodCountBasedOnPrometheusQuery
+
+		if desiredState.EnableScaleDownRatioDeploymentChecking != "true" || !isDeploymentInProgress(kubeClient, hpa, replicaSets) {
```

Comment:
Hmm, I don't think so, because for `hpa`s which don't have this annotation (or have it with the value `"false"`), the first part of the expression (`desiredState.EnableScaleDownRatioDeploymentChecking != "true"`) will be true, so the second part (`!isDeploymentInProgress(kubeClient, hpa, replicaSets)`) will not be evaluated. Or am I mistaken?

But I definitely agree that this is hard to read, I'll change it to sth more straightforward.

Reply:
Changed.

Reply:
Ah my bad, it of course evaluates to true if the value is false, skipping the need to evaluate the second condition.